### PR TITLE
Bug 1131969 - Cache the visual data to reduce the sync reflow when showing the

### PR DESCRIPTION
**Pull request by RudyLu**
keyboard again.

### CODE REVIEW

**Comment by try-server-hook** [comment]
[RudyLu Rudy Lu (RudyLu)](https://github.com/RudyLu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=62c30eadc532)

**Comment by try-server-hook** [comment]
[RudyLu Rudy Lu (RudyLu)](https://github.com/RudyLu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=fc9b61d44f6c)

**Comment by try-server-hook** [comment]
[RudyLu Rudy Lu (RudyLu)](https://github.com/RudyLu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cd1f45ba1836)

**Comment by try-server-hook** [comment]
[RudyLu Rudy Lu (RudyLu)](https://github.com/RudyLu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cc849240dfd9)

**Comment by try-server-hook** [comment]
[RudyLu Rudy Lu (RudyLu)](https://github.com/RudyLu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=77bef130c442)

File: apps/keyboard/test/unit/views/layout_page_view_test.js [hunk 1]
```diff
@@ -355,4 +355,110 @@ suite('Views > LayoutPageView', function() {
...
+
+      pageView = new LayoutPageView(layout, {}, viewManager);
+
+      rootElement = document.createElement('div');
+      document.body.appendChild(rootElement);
+    });
+
+    teardown(function() {
+      document.body.removeChild(rootElement);
+    });
```
Comment (timdream) [inline apps/keyboard/test/unit/views/layout_page_view_test.js]: Do we really need the setup/teardown?

**Comment by try-server-hook** [comment]
[RudyLu Rudy Lu (RudyLu)](https://github.com/RudyLu) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=171c0f9de85c)

**Pull request merged**